packages/cli/internal/pkg/cli/context_deploy.go

```diff
@@ -89,45 +84,94 @@ func (o *deployContextOpts) validateSuppliedContexts(contextList []string) error
 // Execute causes the specified context(s) to be deployed.
 func (o *deployContextOpts) Execute() ([]context.Detail, error) {
 	o.contexts = slices.DeDuplicateStrings(o.contexts)
-	results := o.deployContexts(o.contexts)
-	contextDetails := make([]context.Detail, len(results))
-	hasErrors := false
+
+	err := o.deployContexts()
+	if err != nil {
+		return nil, err
+	}
+
+	contextDetails, err := o.validateDeploymentResults()
+	if err != nil {
+		return nil, err
+	}
+
+	sortContextDetails(contextDetails)
+	return contextDetails, nil
+}
+
+func (o *deployContextOpts) deployContexts() error {
+	progressResults := o.ctxManagerFactory().Deploy(o.contexts)
 	aggregateSuggestions := make([]string, 0)
-	for i, result := range results {
-		if result.err != nil {
+
+	failedDeployments := make([]context.ProgressResult, 0)
```

Comment:
length isn't needed here as a second arg in `make` if it's just going to be zero - go will use sensible defaults